apis/v1alpha1/tcproute_types.go

```diff
@@ -23,6 +23,11 @@ import (
 type TCPRouteSpec struct {
 	// Rules are a list of TCP matchers and actions.
 	Rules []TCPRouteRule `json:"rules"`
+
+	// Gateways defines which Gateways can use this Route.
+	// +kubebuilder:default={allow: "SameNamespace"}
+	// +optional
+	Gateways *RouteGateways `json:"gateways,omitempty"`
```

Comment:
Please add this to UDPRoute and  TLSRoute.

Reply:
Updated, thanks for catching this!